crates/kitsune_p2p/fetch/src/pool.rs

```diff
@@ -201,7 +187,7 @@ impl State {
                 v.sources.0.insert(0, SourceRecord::new(source));
                 v.context = match (v.context.take(), context) {
                     (Some(a), Some(b)) => Some(config.merge_fetch_contexts(*a, *b).into()),
-                    (a, b) => a.and(b),
+                    _ => None,
```

Comment:
One or both are None here so `a.and(b)` will be `None`. May as well just use a literal for readability

Reply:
Technically, according to the docs `a.and(b)` will return `b` if `a` is `None`. Not sure that API makes a whole lot of sense, but that means both the original code and your change are incorrect. We actually want to keep the context if either of the contexts are `Some`, merging them if the are both `Some`.

Reply:
Good spot yes, I've made this explicit and made sure each case is covered by a test